### DISCUSSION TITLE
Custom drawing & bounds

### DISCUSSION THREAD

**Discussion by mgnslndh** [AvaloniaUI/Avalonia#12949] [Q&A]
I am writing a custom control that draw itself using a ICustomDrawOperation. 

- The control knows the size of the drawing. Where is the correct place to assign this size? I'm using MeasureOverride to return the desired size.

- The custom draw operation has a Bounds property and I assign it a rectangle located at 0,0 with the same size as I used in the MeasureOverride method.

1. Is this the correct procedure?
2. What would it mean to have bounds that are offset from 0,0 on the operation? Is it OK to draw outside of the control bounds?

**Comment by timunie** [comment]
@mgnslndh check for example what Image does. Haven't tried to render outside bounds, but I guess it depends on wether ClipToBounds is set. probably you need to specify dirty rects in that case (don't know tbh)
